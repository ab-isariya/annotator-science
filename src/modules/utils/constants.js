import {colors} from '@ui/theme';
import {ReactComponent as IconDocx} from '@assets/svgs/IconDocx.svg';
import {ReactComponent as IconPdf} from '@assets/svgs/IconPdf.svg';

/*
 * Entity/Annotation Status enum
 * See tat-api/annotate_api/chalicelib/database/utils.py
 */
export const AnnotationStatus = {
  ACCEPTED: 'ACCEPTED',
  NOT_REVIEWED: 'NOT_REVIEWED', //DEFAULT
  REJECTED: 'REJECTED',
  MANUAL: 'MANUAL'
};

/** Labels for Entity/Annotation status' */
export const AnnotationStatusLabel = {
  ACCEPTED: 'Accepted',
  REJECTED: 'Rejected',
  NOT_REVIEWED: 'To Review',
  MANUAL: 'Added'
};

//NOTE(Rejon): Progress Type aggregations aren't sorted on the backend,
//             this array also helps controls the expected order for Progress Types
export const sortedAnnotationStatus = [
  AnnotationStatus.ACCEPTED,
  AnnotationStatus.REJECTED,
  AnnotationStatus.NOT_REVIEWED,
  AnnotationStatus.MANUAL
];

/** Color codes specifically used for PieChart components. */
export const annotationStatusColors = {
  ACCEPTED: colors.green['400'],
  REJECTED: colors.red['600'],
  NOT_REVIEWED: colors.grey['200'],
  MANUAL: colors.blue['500'],
  ZERO: colors.grey['50'] // no annotations
};

// Color codes for statuses rendered using only Text
export const altAnnotationStatusColors = {
  ACCEPTED: 'text-green-500',
  REJECTED: 'text-red-600',
  NOT_REVIEWED: 'text-grey-500',
  MANUAL: 'text-black'
};

//Annotation type tags based on Kepler spec.
export const annotationTypes = {
  ANATOMY: 'anatomy & physiology',
  CELLBIOLOGY: 'cell biology',
  CHEMICALS: 'chemicals & drugs',
  DEVICES: 'medical devices',
  CONDITIONS: 'medical conditions',
  GENETICS: 'genetics',
  PROCEDURES: 'medical procedures',
  VIRUSES: 'species & viruses',
  CONTEXT: 'context',
  UNASSIGNED: 'unassigned'
};

/*
 * Document processing state
 * See tat-api/annotate_api/chalicelib/database/utils.py
 *
 * Sorted values used in src/modules/document/table/sortFunctions.js
 */
export const ProcessingState = {
  CREATED: 'CREATED',
  UPLOADED: 'UPLOADED',
  QUEUED: 'QUEUED',
  FAILED: 'FAILED',
  PROCESSING: 'PROCESSING',
  PROCESSED: 'PROCESSED'
};

//Processing State sorted
export const sortedProcessingStatus = [
  'CREATED',
  'UPLOADED',
  'QUEUED',
  'PROCESSING',
  'PROCESSED'
];

/*
 * Annotation confidence scores based on Kepler scoring rubric
 */
export const ConfidenceScore = {
  very_high: {
    label: 'Very High',
    max: 1,
    min: 0.99,
    tooltipText: 'Very High Confidence',
    score_cat: 'VERY_HIGH'
  },
  high: {
    label: 'High',
    max: 0.98,
    min: 0.9,
    tooltipText: 'High Confidence',
    score_cat: 'HIGH'
  },
  moderate: {
    label: 'Moderate',
    max: 0.89,
    min: 0.7,
    tooltipText: 'Moderate Confidence',
    score_cat: 'MODERATE'
  },
  low: {
    label: 'Low',
    max: 0.69,
    min: 0.5,
    tooltipText: 'Low Confidence',
    score_cat: 'LOW'
  },
  very_low: {
    label: 'Very Low',
    max: 0.49,
    min: 0,
    tooltipText: 'Very Low Confidence',
    score_cat: 'VERY_LOW'
  }
};

//NOTE(Rejon): Confidence Type aggregations don't sort on the backend,
//             this controls the expected sort order if confidence type
//             is ever listed together.
export const ConfidenceTypeOrder = [
  'very_high',
  'high',
  'moderate',
  'low',
  'very_low'
];

// Keys that correspond to the backend model for filtering Annotations
// based on the client side modules they're used in.
export const AnnotationFilterKeys = {
  ModelConfidence: 'entity_p',
  Progress: 'status',
  Types: 'tag',
  Concepts: 'canonical_id'
};

//Possible view states for the document view.
export const DocumentViewTypes = {
  TABLE: 'table',
  DOCUMENT: 'document'
};

//Possible Document Export table options
export const DocumentExportTypes = {
  TSV: 'TSV',
  CSV: 'CSV',
  XLSX: 'XLSX'
};

//Lenght that a snippet should be.
export const SNIPPET_CLAMP = 25;

//Right sidebar view modes.
export const RightSidebarModes = {
  REVIEW: 'REVIEW',
  ADD: 'ADD',
  EDIT: 'EDIT'
};

//Supported upload file type icons.
export const FileTypes = {
  'docx': IconDocx,
  'pdf': IconPdf
};
