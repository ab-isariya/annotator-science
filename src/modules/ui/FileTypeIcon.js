import {createElement} from 'react';
import PropTypes from 'prop-types';

import {FileTypes} from '@utils/constants';

const FileTypeIcon = ({type, className}) => {
  const _type = type.replace('.', '');
  const icon = FileTypes[_type];

  if (!icon) {
    return <></>;
  }

  return createElement(FileTypes[_type], {
    className
  });
};

FileTypeIcon.propTypes = {
  //Tailwind classes
  className: PropTypes.string,
  //Filetype string
  type: PropTypes.oneOf(Object.keys(FileTypes).map(e => e.replace('.', ''))).isRequired
};

export default FileTypeIcon;
