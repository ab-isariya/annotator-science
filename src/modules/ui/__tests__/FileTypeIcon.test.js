import FileTypeIcon from '../FileTypeIcon';
import {FileTypes} from '@utils/constants';

import renderer from 'react-test-renderer';

describe('<FileTypeIcon/>', () => {
  it('Renders without crashing', () => {
    const comp = renderer.create(<FileTypeIcon type={'.docx'} />).toJSON();

    expect(comp).toMatchSnapshot();
  });

  it('Renders with a custom classname', () => {
    const comp = renderer
      .create(<FileTypeIcon type={'.pdf'} className="testClass" />)
      .toJSON();

    expect(comp).toMatchSnapshot();
    expect(comp.props.className).toMatch(/testClass/i);
  });
});
