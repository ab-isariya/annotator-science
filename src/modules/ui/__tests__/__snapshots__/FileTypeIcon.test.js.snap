// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FileTypeIcon/> Renders with a custom classname 1`] = `
<svg
  className="testClass"
>
  IconPdf.svg
</svg>
`;

exports[`<FileTypeIcon/> Renders without crashing 1`] = `
<svg>
  IconDocx.svg
</svg>
`;
